### PR TITLE
chore(Storybook): refactor Checkbox, RadioButton, Switch and Select

### PR DESCRIPTION
**Pull request by marcoskolodny**
Issue: [Link](https://jira.tid.es/browse/WEB-819)

### CODE REVIEW

**Review by yceballost — CHANGES_REQUESTED** [review]

File: src/__stories__/checkbox-story.tsx
[the remark this answers is not in the document]
Comment (marcoskolodny) [reply src/__stories__/checkbox-story.tsx:1]: I've updated Checkbox/RadioButton/Switch custom render stories so that all of them render custom icons instead of default controls.

File: src/__screenshot_tests__/__image_snapshots__/checkbox-screenshot-test-tsx-checkbox-uncontrolled-1-snap.png
Comment (yceballost) [inline src/__screenshot_tests__/__image_snapshots__/checkbox-screenshot-test-tsx-checkbox-uncontrolled-1-snap.png:1]: is it expected the dimensions of this screenshot? because in select screenshot the container fits to the component, for example

**Review by yceballost — APPROVED** [review]

File: src/__screenshot_tests__/__image_snapshots__/checkbox-screenshot-test-tsx-checkbox-control-element-aligned-correctly-with-long-content-1-snap.png
Comment (pladaria) [inline src/__screenshot_tests__/__image_snapshots__/checkbox-screenshot-test-tsx-checkbox-control-element-aligned-correctly-with-long-content-1-snap.png:1]: I'd take the screenshot of the component, not the full page

Comment (marcoskolodny) [reply src/__screenshot_tests__/__image_snapshots__/checkbox-screenshot-test-tsx-checkbox-control-element-aligned-correctly-with-long-content-1-snap.png:1]: fixed

**Review by pladaria — APPROVED** [review]

**Review by pladaria — APPROVED** [review]

File: src/switch-component.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ const Switch: React.FC<PropsRender | PropsChildren> = (props) => {
     return (
         // When the switch is disabled, it shouldn't be focusable
         // eslint-disable-next-line jsx-a11y/interactive-supports-focus
-        <span
+        <div
```
Comment (atabel) [inline src/switch-component.tsx:124]: why this change? I'm afraid this change could be problematic

Comment (marcoskolodny) [reply src/switch-component.tsx]: You're right. Instead of using div, I added `display: inline-block` to the span

**Review by atabel — APPROVED** [review]
